Modules/rat_board.py

```diff
@@ -199,10 +199,39 @@ def find_by_uuid(self, guid: UUID) -> Optional[Rescue]:
 
         """
         for rescue in self.rescues.values():
-            if rescue.case_id == guid:
+            if rescue.uuid == guid:
                 return rescue
         return None
 
+    def search(self, case: str) -> Optional[Rescue]:
+        """
+        Helper for searching for a case using implemented search functions.
+
+        Args:
+            case str: The case string to search on
+
+        Returns:
+            Rescue: found rescue
+            None:   No rescue found
+
+        Raises:
+            ValueError: argument was not an expected type
+        """
+        # ensure a proper type was passed
+        if not isinstance(case, str):
+            raise ValueError(f"Expected str, got {type(case)}")
```

Comment:
This should be a `TypeError`, as its the type that is the problem.

`ValueError` is for when the value is the correct type, but isn;t within the acceptable value range